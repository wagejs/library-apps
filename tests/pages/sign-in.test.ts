import { describe, test, expect, vi } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import SignInPage from '@pages/auth/sign-in.vue'

vi.mock('useAuth', () => ({
  signIn: vi.fn(),
}))

describe('SignInPage', () => {
  test('should render the sign in page', async () => {
    const wrapper = await mountSuspended(SignInPage)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
