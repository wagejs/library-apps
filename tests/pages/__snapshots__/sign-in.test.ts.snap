// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInPage > CASE: Rendering > should render the sign in page correctly 1`] = `
"<div class="flex flex-col justify-center items-center h-screen bg-gray-100">
  <h2 class="text-2xl font-bold mb-8">Sign in to your account</h2>
  <div class="flex flex-col gap-4 max-w-md w-full overflow-hidden shadow-lg mb-4 rounded-lg bg-white p-8">
    <form id="sign-in-form" class="space-y-4 w-full">
      <div class="text-base w-full" id="email-wrapper">
        <div class="">
          <div class="flex content-center items-center justify-between"><label for="email-input" class="block font-medium text-default after:content-['*'] after:ms-0.5 after:text-error"><label for="email" id="email-label" class="text-md font-medium text-gray-700 mb-1">Email</label></label>
            <!--v-if-->
          </div>
          <!--v-if-->
        </div>
        <div class="mt-1 relative">
          <div class="relative inline-flex items-center w-full"><input id="email-input" type="text" class="w-full rounded-md border-0 placeholder:text-dimmed focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors px-3 py-2 text-base gap-2 text-highlighted bg-default ring ring-inset ring-accented focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" autocomplete="off" aria-describedby="" aria-invalid="false" value="">
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
        </div>
      </div>
      <div class="text-base" id="password-wrapper">
        <div class="">
          <div class="flex content-center items-center justify-between"><label for="password-input" class="block font-medium text-default after:content-['*'] after:ms-0.5 after:text-error"><label for="password" id="password-label" class="text-md font-medium text-gray-700 mb-1">Password</label></label>
            <!--v-if-->
          </div>
          <!--v-if-->
        </div>
        <div class="mt-1 relative">
          <div class="relative inline-flex items-center w-full"><input id="password-input" type="password" class="w-full rounded-md border-0 placeholder:text-dimmed focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors px-3 py-2 text-base gap-2 text-highlighted bg-default ring ring-inset ring-accented focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary" autocomplete="off" aria-describedby="" aria-invalid="false" value="">
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
        </div>
      </div><button type="submit" id="sign-in-button" class="rounded-md disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 text-base gap-2 w-full text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary flex font-bold justify-center items-center mt-8 py-3">
        <!--v-if--><span class="truncate">Sign in</span><span class="iconify i-heroicons:arrow-right-circle" aria-hidden="true" style="font-size: 20px;"></span>
      </button>
    </form>
    <div class="flex gap-2 mt-2 justify-center"><span>Don't have an account?</span><a href="/auth/sign-up" class="focus-visible:outline-primary hover:text-default transition-colors text-green-700">Sign up</a></div>
  </div>
</div>"
`;
